moveit_core/collision_detection/src/world.cpp

```diff
@@ -136,6 +136,95 @@ bool World::hasObject(const std::string& object_id) const
   return objects_.find(object_id) != objects_.end();
 }
 
+bool World::knowsTransform(const std::string& object_id) const
+{
+  // Check object names first
+  if (objects_.find(object_id) != objects_.end())
+    return true;
+  else  // Then objects' subframes
+  {
+    if (object_id.find("/")!=std::string::npos)
+    {
+      for (const std::pair<std::string, ObjectPtr>& object : objects_)
+      {
+        try 
+        {
+          // Compare both object name and subframe name
+          int name_length = object.second->id_.length();
+          if (object.second->id_ == object_id.substr(0,name_length))
+          {
```

Comment:
This code continues to search the (remaining) object list, even if the object name matched, but a corresponding subframe wasn't present.
Assuming that we don't have so many free objects in a scene, I accept this linear search through the object list for now.

Reply:
True. I'll add a break.

Reply:
Fixed in my fork.